### PR TITLE
Build status API end point

### PR DESCRIPTION
**Pull request by CorneliusIV**

### CODE REVIEW

File: franklin/builder/models.py [hunk 1]
```diff
@@ -206,21 +204,22 @@ class Environment(models.Model):
...
-                        "repo_name": self.site.name
-                    }
+                "deploy_key": os.environ['GITHUB_OAUTH'],
+                "branch": self.current_deploy.branch if not is_tag_build else '',
+                "tag": self.current_deploy.tag if is_tag_build else '',
+                "git_hash": self.current_deploy.git_hash if not is_tag_build else '',
+                "repo_owner": self.site.owner.name,
+                "path": self.current_deploy.path,
+                "repo_name": self.site.name,
+                "environment_id": self.id
```
Comment (CorneliusIV) [inline franklin/builder/models.py]: This is mostly PEP 8 stuff the environment_id is the only change I made

File: franklin/builder/serializers.py [hunk 2]
```diff
@@ -27,6 +27,18 @@ class Meta:
         }
 
 
+class EnvironmentStatusSerializer(serializers.ModelSerializer):
```
Comment (CorneliusIV) [inline franklin/builder/serializers.py]: I'm not sure best practices, if you you are supposed to have one Serializer for one model, or different Serializers for different functions.  I have seen a mixture of both.

**Comment by JulianGindi** [comment]
This looks good to me :+1:

File: franklin/builder/serializers.py [hunk 3]
```diff
@@ -27,6 +27,18 @@ class Meta:
...
+
+    class Meta:
+        model = Environment
+        fields = ('status',)
+        extra_kwargs = {
+            "github_id": {
+                "validators": [],
+            },
+        }
```
Comment (oudeismetis) [inline franklin/builder/serializers.py]: extra_kwargs not needed here. We did it above due to a bug. What this is trying to do is look for `github_id` on the model and tell the serializer to skip validation for it. You don't have that on this model.

File: franklin/builder/views.py [hunk 4]
```diff
@@ -1,3 +1,26 @@
...
+
+from .serializers import EnvironmentStatusSerializer
+
+logger = logging.getLogger(__name__)
+
+
+class UpdateBuildStatus(generics.UpdateAPIView):
+    queryset = Environment.objects.all()
+    serializer_class = EnvironmentStatusSerializer
+    permission_classes = (AllowAny,)
```
Comment (oudeismetis) [inline franklin/builder/views.py]: I assume we are just skipping security on this endpoint for the moment and will eventually create special access that only allows builder to call this.

**Comment by oudeismetis** [comment]
Other than my one comment, this looks real good. :+1:

File: franklin/builder/views.py (hunk 4 above)
Comment (CorneliusIV) [inline franklin/builder/views.py]: I was thinking the same thing.

File: franklin/builder/serializers.py (hunk 3 above)
Comment (CorneliusIV) [inline franklin/builder/serializers.py]: Cool, I will update it, wasn't sure if it was needed or not.

**Pull request merged**